backend/src/layer/data/database/request.rs

```diff
@@ -286,6 +286,54 @@ impl RequestDataAccess for PersistentRequestData {
         .await?;
         Ok(res)
     }
+
+    async fn get_nutrition_data(&self, food_id: Uuid) -> Result<Option<NutritionData>> {
+        let res = sqlx::query!(
+            r#"SELECT energy, protein, carbohydrates, sugar, fat, saturated_fat, salt FROM food_nutrition_data WHERE food_id = $1"#,
+            food_id
+        ).fetch_optional(&self.pool)
+        .await?;
+        let result = match res {
+            Some(res) => Some(NutritionData {
+                energy: u32::try_from(res.energy)?,
+                protein: u32::try_from(res.protein)?,
+                carbohydrates: u32::try_from(res.carbohydrates)?,
+                sugar: u32::try_from(res.sugar)?,
+                fat: u32::try_from(res.fat)?,
+                saturated_fat: u32::try_from(res.saturated_fat)?,
+                salt: u32::try_from(res.salt)?,
+            }),
+            None => None,
+        };
+        Ok(result)
+    }
+
+    async fn get_environment_information(&self, food_id: Uuid) -> Result<Option<EnvironmentInfo>> {
+        let res = sqlx::query!(
+            r#"SELECT co2_rating, co2_value, water_rating, water_value, animal_welfare_rating, rainforest_rating, max_rating FROM food_env_score WHERE food_id = $1"#,
+            food_id
+        ).fetch_optional(&self.pool).await?;
+        if let Some(res) = res {
+            let co2_rating = u32::try_from(res.co2_rating)?;
+            let water_rating = u32::try_from(res.water_rating)?;
+            let animal_welfare_rating = u32::try_from(res.animal_welfare_rating)?;
+            let rainforest_rating = u32::try_from(res.rainforest_rating)?;
+            let average_rating =
+                (co2_rating + water_rating + animal_welfare_rating + rainforest_rating) / 4;
```

Comment:
does that correctly get converted to f32?

Reply:
No, this should be an unsigned integer, like the original.

Reply:
Never mind, Github showed me the wrong type for average_rating (the one from Meal). I should trust the type system 😁